src/components/Authz/Credentials.js

```diff
@@ -16,152 +14,57 @@ export function Credentials({ ...props }) {
 	const resources = useSelector(state => state?.auth?.resources);
 	const resource = 'seacat:credentials:access'; // Resource required to access the Credentials List Screen
 
-	const { t } = useTranslation();
-
-	const apiPath = props.apiPath ?? 'seacat-auth';
-
-	const credentials_ids = Array.isArray(props.credentials_ids) ? props.credentials_ids : [props.credentials_ids];
-
 	// Validation on props.app
 	if (props.app == undefined) {
-		return renderPlainCredentials(credentials_ids);
+		return renderPlainCredentials(props.credentials_ids);
 	}
 
-	const CredentialsAPI = props.app.axiosCreate(apiPath);
-
 	const cleanupTime = props.cleanupTime ?? 1000 * 60 * 60 * 24; // 24 hrs
 
-	const [credentials, setCredentials] = useState([]);
+	const [credential, setCredential] = useState(null);
 	// Checks if there is a SeaCatAdminFederationModule
 	const hasSeaCatAdminModule = props.app?.Modules.some((obj) => obj.Name === 'SeaCatAdminFederationModule') === false;
 
 	useEffect(() => {
-		matchCredentialIds(credentials_ids);
+		matchCredentialId(props.app, props.credentials_ids, setCredential, cleanupTime);
```

Comment:
@aringocode since we use (almost everywhere) a property `credentials_ids` which is an `array` `[]` - you really need to add a fallback. Also I would recommend to do it this way:


```
const credentials = props.credentials_id ? props.credentials_id : props.credentials_ids?.[0];
setCredential(credentials);
```
by doing it so, you will ensure the fallback to previous implementation (you can gracefully change `credentials_ids=["some credential id"]` into `credentials_id="some credential id"` )
by doing it so, you ensure a fallback

Reply:
@Pe5h4 
I fixed it. I think fallback is better to add for Credentals because it is needed there. I will not add it for User, because it is a new component. I will make a demo for Credentials and User. 
Can you please still give answers to the questions I wrote above.
https://github.com/TeskaLabs/asab-webui-components-lib/pull/14#discussion_r2024715145
https://github.com/TeskaLabs/asab-webui-components-lib/pull/14#discussion_r2024713616
https://github.com/TeskaLabs/asab-webui-components-lib/pull/14#discussion_r2024710796